servers/src/main/java/tachyon/master/file/meta/TTLBucket.java

```diff
@@ -80,4 +82,26 @@ public int compareTo(TTLBucket ttlBucket) {
     }
     return 1;
   }
+
+  /**
+   * Compares to a specific object.
+   *
+   * @param object the object to compare
+   * @return true if object is also {@link TTLBucket} and represents
+   * the same TTLIntervalStartTime
+   */
+  @Override
+  public boolean equals(Object object) {
+    if (object == null || getClass() != object.getClass()) {
+      return false;
+    }
+
+    TTLBucket that = (TTLBucket) object;
+    return Objects.equal(mTTLIntervalStartTimeMs, that.mTTLIntervalStartTimeMs);
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hashCode(mTTLIntervalStartTimeMs, mFiles);
```

Comment:
Does this mean the hashCode would incorporate both the <code>mTTLIntervalStartTimeMs</code> and <code>mFiles</code>? This means 2 buckets with the same start time, but different mFiles, would have a different hashCode(), but equals() would be true. I think that violates the contract between equals() and hashCode().